lib/ood_core/job/adapters/slurm.rb

```diff
@@ -643,7 +643,8 @@ def parse_job_info(v)
               cpu_time: nil,
               submission_time: v[:submit_time] ? Time.parse(v[:submit_time]) : nil,
               dispatch_time: (v[:start_time].nil? || v[:start_time] == "N/A") ? nil : Time.parse(v[:start_time]),
-              native: v
+              native: v,
+              gpus: v[:gres]&.include?("gpu") ? 1 : 0
```

Comment:
We probably want some sort of regular expression here to extract a number to get the right number.

Reply:
The format for GRES is `gres:subtype:number` so if you split on colons and get last element, that should be the number.  I would not recommend extracting digits because at OSC, we have numbers in GRES subtype like we have `gpu:v100:2` or `gpu:v100-32g:2`.  Maybe `([^:]+):([^:]+):([0-9]+)` if you really want to use regex.

Reply:
Ah my regex is a little too simple, the `subtype` is optional so you can have `gres:number` or `gres:subtype:number`.  So I think in those cases just splitting on colons and pulling out the last element might be simplest to avoid overly complex regex.

Reply:
We'll need at least the `gpu` to check against others like `pfsdir` and so on.  

Maybe `[,]*gpu.*:(\d+)[,]*`?  @lukew3 I'll update this ticket for test cases we want to be sure to cover.

Reply:
> Maybe `[,]*gpu.*:(\d+)[,]*`? @lukew3 I'll update this ticket for test cases we want to be sure to cover.

I split by commas and colons to count it, so this works:
```ruby
gres&.split(',')&.each { |s| gpus += s.split(':').last.to_i if s.split(':')[0] == 'gpu' }
```